hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestRegionServerReportForDuty.java

```diff
@@ -241,6 +243,125 @@ public void run() {
+   */
+  @Test
+  public void testReportForDutyGetsAcceptedByMasterWhenNotConfiguredToRejectDecommissionedHosts()
+    throws Exception {
+    // Start a master and wait for it to become the active/primary master.
+    // Use a random unique port
+    cluster.getConfiguration().setInt(HConstants.MASTER_PORT, HBaseTestingUtil.randomFreePort());
+    cluster.getConfiguration().setInt(ServerManager.WAIT_ON_REGIONSERVERS_MINTOSTART, 1);
+    cluster.getConfiguration().setInt(ServerManager.WAIT_ON_REGIONSERVERS_MAXTOSTART, 1);
+
+    // Set the cluster to not reject decommissioned hosts (default behavior)
+    cluster.getConfiguration().setBoolean(HConstants.REJECT_DECOMMISSIONED_HOSTS_KEY, false);
+
+    master = cluster.addMaster();
+    rs = cluster.addRegionServer();
+    LOG.debug("Starting master: " + master.getMaster().getServerName());
+    master.start();
+    rs.start();
+
+    waitForClusterOnline(master);
+
+    assertEquals(0, master.getMaster().listDecommissionedRegionServers().size());
+    assertEquals(0, master.getMaster().getServerManager().getDrainingServersList().size());
+    assertEquals(1, master.getMaster().getServerManager().getOnlineServers().size());
+
+    // Decommission the region server and tries to re-add it
+    List<ServerName> serversToDecommission = new ArrayList<ServerName>();
+    serversToDecommission.add(rs.getRegionServer().getServerName());
+    master.getMaster().decommissionRegionServers(serversToDecommission, true);
+
+    // Assert that the server is now decommissioned
+    ServerName decommissionedServer = master.getMaster().listDecommissionedRegionServers().get(0);
+    assertEquals(1, master.getMaster().listDecommissionedRegionServers().size());
+    assertEquals(1, master.getMaster().getServerManager().getDrainingServersList().size());
+    assertEquals(1, master.getMaster().getServerManager().getOnlineServers().size());
+    assertEquals(rs.getRegionServer().getServerName().toString(),
+      decommissionedServer.getServerName());
+
+    // Create a second region server
+    cluster.getConfiguration().set(HConstants.REGION_SERVER_IMPL, MyRegionServer.class.getName());
+    rs2 = cluster.addRegionServer();
+    rs2.start();
+    waitForSecondRsStarted();
+    master.getMaster().recommissionRegionServer(rs2.getRegionServer().getServerName(),
+      new ArrayList<>());
+
+    // Assert that the number of decommissioned hosts is the same and that the live and online hosts
+    // did in fact change and rs2 is now part of the cluster even though there is a match for its
+    // hostname in the list of decommissioned servers
+    String rs2HostName = rs2.getRegionServer().getServerName().getHostname();
+    assertEquals(1, master.getMaster().listDecommissionedRegionServers().size());
+    assertEquals(1, master.getMaster().getServerManager().getDrainingServersList().size());
+    assertEquals(2, master.getMaster().getLiveRegionServers().size());
```

Comment:
This test is failing on this line. This is the case when the cluster is NOT configured to reject decommissioned hosts. It fails due to `2 != 1`, I don't get how the first few assertions passed if this second one is not. I debugged it and it seems like the list of LiveRegionServers and OnlineRegionServers don't have the second RegionServer `rs2` in them.